processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java

```diff
@@ -369,13 +374,13 @@ public boolean doMergeResults(final GroupByQuery query)
                    .map(AggregatorFactory::getCombiningFactory)
                    .collect(Collectors.toList())
           )
-          .withSubtotalsSpec(null)
-          .withDimFilter(null);
-
+          .withVirtualColumns(VirtualColumns.EMPTY)
+          .withDimFilter(null)
+          .withSubtotalsSpec(null);
```

Comment:
What do you mean by 'make sense'? It makes sense to me in that you have an immutable object (the GroupByQuery) and you are using chaining to create a derived copy.

Reply:
Sorry, I should have phrased that better.

I meant that there seems to be a lot of overlap between the `.with...` functions and the `GroupByQuery.Builder` functions. When I saw that pattern, I was wondering how do you choose between the 2. And since both exist, I may not pick the correct one.

Nothing to change here - just my stream of thought while reading the code.